crates/forge/tests/integration/declare.rs

```diff
@@ -106,21 +109,25 @@ fn redeclare() {
     let test = test_case!(
         indoc!(
             r#"
+        use core::clone::Clone;
         use result::ResultTrait;
         use traits::Into;
         use starknet::ClassHashIntoFelt252;
-        use snforge_std::declare;
+        use snforge_std::{declare, DeclareResultTrait, DeclareResult};
 
         #[test]
         fn redeclare() {
             assert(1 == 1, 'simple check');
             let contract = match declare("HelloStarknet") {
-                Result::Ok(contract) => contract,
+                Result::Ok(result) => result.contract_class().clone(),
                 Result::Err(_) => panic!("Failed to declare contract"),
             };
             let class_hash = contract.class_hash.into();
             assert(class_hash != 0, 'proper class hash');
-            assert!(declare("HelloStarknet").is_err(), "Contract redeclared");
+            match declare("HelloStarknet").unwrap() {
+                DeclareResult::AlreadyDeclared(_) => {},
+                _ => panic!("Contract redeclared")
```

Comment:
I think it should be that? 

```suggestion
                _ => panic!("Contract not redeclared")
```

Reply:
I think it is okay as it is. Re-declaration is a mistake, so it should panic with `Contract redeclared`.